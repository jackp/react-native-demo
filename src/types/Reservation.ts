export interface Reservation {
  id: string;
  name: string;
  hotelName: string;
  arrivalDate: string;
  departureDate: string;
}
