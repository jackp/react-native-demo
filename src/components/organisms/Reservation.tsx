/**
 * <Reservation />
 * - Display reservation information
 */

import React from "react";
import { Card, CardItem, Text, Title, Body } from "native-base";

import { Reservation as ReservationType } from "../../types/Reservation";

interface IProps {
  reservation: ReservationType;
}

const Reservation: React.FunctionComponent<IProps> = ({ reservation }) => (
  <Card transparent={true} style={{ flex: 1 }}>
    <CardItem header={true}>
      <Title testID="name">{reservation.name}</Title>
    </CardItem>
    <CardItem>
      <Body>
        <Text testID="hotelName">Hotel: {reservation.hotelName}</Text>
        <Text testID="arrivalDate">
          Arrival Date: {reservation.arrivalDate}
        </Text>
        <Text testID="departureDate">
          Departure Date: {reservation.departureDate}
        </Text>
      </Body>
    </CardItem>
  </Card>
);

export default Reservation;
