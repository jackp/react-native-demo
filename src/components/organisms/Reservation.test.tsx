import "react-native";
import React from "react";
import { mount } from "enzyme";

import { Reservation as ReservationType } from "../../types/Reservation";
import Reservation from "./Reservation";

const reservation: ReservationType = {
  id: "1",
  name: "Jack Parker",
  hotelName: "Hilton PVD",
  arrivalDate: "3/7/2019",
  departureDate: "3/17/2019"
};

describe("<Reservation />", () => {
  it("displays all relevant data", () => {
    const wrapper = mount(<Reservation reservation={reservation} />);
    expect(wrapper).toMatchSnapshot();

    expect(
      wrapper.findWhere(node => node.prop("testID") === "name").first()
    ).toIncludeText(reservation.name);
    expect(
      wrapper.findWhere(node => node.prop("testID") === "hotelName").first()
    ).toIncludeText(reservation.hotelName);
    expect(
      wrapper.findWhere(node => node.prop("testID") === "arrivalDate").first()
    ).toIncludeText(reservation.arrivalDate);
    expect(
      wrapper.findWhere(node => node.prop("testID") === "departureDate").first()
    ).toIncludeText(reservation.departureDate);
  });
});
