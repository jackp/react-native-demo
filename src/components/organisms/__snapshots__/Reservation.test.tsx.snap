// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reservation /> displays all relevant data 1`] = `
<Reservation
  reservation={
    Object {
      "arrivalDate": "3/7/2019",
      "departureDate": "3/17/2019",
      "hotelName": "Hilton PVD",
      "id": "1",
      "name": "Jack Parker",
    }
  }
>
  <Styled(Card)
    style={
      Object {
        "flex": 1,
      }
    }
    transparent={true}
  >
    <Card
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomWidth": 0,
            "borderColor": "#ccc",
            "borderLeftWidth": 0,
            "borderRadius": 2,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "elevation": null,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          },
          Object {
            "flex": 1,
          },
        ]
      }
      transparent={true}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomWidth": 0,
              "borderColor": "#ccc",
              "borderLeftWidth": 0,
              "borderRadius": 2,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "elevation": null,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            },
            Object {
              "flex": 1,
            },
          ]
        }
        transparent={true}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderColor": "#ccc",
                "borderLeftWidth": 0,
                "borderRadius": 2,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "elevation": null,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              },
              Object {
                "flex": 1,
              },
            ]
          }
          transparent={true}
        >
          <Styled(CardItem)
            header={true}
          >
            <CardItem
              header={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": null,
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 15,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                }
              }
            >
              <View
                header={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderBottomWidth": null,
                    "borderRadius": 2,
                    "flexDirection": "row",
                    "paddingBottom": 15,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 15,
                  }
                }
              >
                <View
                  header={true}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderBottomWidth": null,
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "paddingBottom": 15,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "paddingTop": 15,
                    }
                  }
                >
                  <Styled(Title)
                    testID="name"
                  >
                    <Title
                      style={
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 17,
                          "fontWeight": "700",
                          "marginLeft": undefined,
                          "paddingLeft": 4,
                          "paddingTop": 1,
                          "textAlign": "center",
                        }
                      }
                      testID="name"
                    >
                      <Text
                        numberOfLines={1}
                        style={
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 17,
                            "fontWeight": "700",
                            "marginLeft": undefined,
                            "paddingLeft": 4,
                            "paddingTop": 1,
                            "textAlign": "center",
                          }
                        }
                        testID="name"
                      >
                        <Text
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 17,
                              "fontWeight": "700",
                              "marginLeft": undefined,
                              "paddingLeft": 4,
                              "paddingTop": 1,
                              "textAlign": "center",
                            }
                          }
                          testID="name"
                        >
                          Jack Parker
                        </Text>
                      </Text>
                    </Title>
                  </Styled(Title)>
                </View>
              </View>
            </CardItem>
          </Styled(CardItem)>
          <Styled(CardItem)>
            <CardItem
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderRadius": 2,
                    "flexDirection": "row",
                    "paddingBottom": 10,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "paddingBottom": 10,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "paddingTop": 10,
                    }
                  }
                >
                  <Styled(Body)>
                    <Body
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "alignSelf": "stretch",
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "alignSelf": "stretch",
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "stretch",
                              "flex": 1,
                            }
                          }
                        >
                          <Styled(Text)
                            testID="hotelName"
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                }
                              }
                              testID="hotelName"
                              uppercase={false}
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#000",
                                    "fontFamily": "System",
                                    "fontSize": 16,
                                  }
                                }
                                testID="hotelName"
                                uppercase={false}
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "System",
                                      "fontSize": 16,
                                    }
                                  }
                                  testID="hotelName"
                                  uppercase={false}
                                >
                                  Hotel: 
                                  Hilton PVD
                                </Text>
                              </Text>
                            </Text>
                          </Styled(Text)>
                          <Styled(Text)
                            testID="arrivalDate"
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                }
                              }
                              testID="arrivalDate"
                              uppercase={false}
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#000",
                                    "fontFamily": "System",
                                    "fontSize": 16,
                                  }
                                }
                                testID="arrivalDate"
                                uppercase={false}
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "System",
                                      "fontSize": 16,
                                    }
                                  }
                                  testID="arrivalDate"
                                  uppercase={false}
                                >
                                  Arrival Date: 
                                  3/7/2019
                                </Text>
                              </Text>
                            </Text>
                          </Styled(Text)>
                          <Styled(Text)
                            testID="departureDate"
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                }
                              }
                              testID="departureDate"
                              uppercase={false}
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#000",
                                    "fontFamily": "System",
                                    "fontSize": 16,
                                  }
                                }
                                testID="departureDate"
                                uppercase={false}
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "System",
                                      "fontSize": 16,
                                    }
                                  }
                                  testID="departureDate"
                                  uppercase={false}
                                >
                                  Departure Date: 
                                  3/17/2019
                                </Text>
                              </Text>
                            </Text>
                          </Styled(Text)>
                        </View>
                      </View>
                    </Body>
                  </Styled(Body)>
                </View>
              </View>
            </CardItem>
          </Styled(CardItem)>
        </View>
      </View>
    </Card>
  </Styled(Card)>
</Reservation>
`;
